addons/api/addon/serializers/host-catalog.js

```diff
@@ -68,15 +68,29 @@ export default class HostCatalogSerializer extends ApplicationSerializer {
     const value = super.serializeAttribute(...arguments);
     const { compositeType, credentialType, isPlugin } = snapshot.record;
     const { options } = attribute;
-
     const fields =
       compositeType === TYPE_HOST_CATALOG_PLUGIN_AZURE
         ? azureFields
         : AWSfieldsWithCredentialType[credentialType];
 
-    // Delete any nested attribute fields that don't belong to the record type
+    // Delete any nested attribute fields that don't belong to the record's compositeType
     if (isPlugin && options.isNestedAttribute && json.attributes) {
-      if (!fields.includes(key)) {
+      // If compositeType is AWS, delete azure fields from the json, and set the rest that don't belong to respective compositeType to null
+      if (compositeType === TYPE_HOST_CATALOG_PLUGIN_AWS) {
+        if (!AWSfieldsWithCredentialType[credentialType].includes(key)) {
+          if (azureFields.includes(key)) {
+            delete json.attributes[key];
+          } else {
+            json.attributes[key] = null;
```

Comment:
question for understanding: why do the attrs that don't belong to the seleted aws cred. type need to be set to null, why not delete the attr ? Is it because that is what the api expects ?

Reply:
yeah in patch calls, we need to send null otherwise API won't know if we updated that field. 
For instance, if we have a dynamic credential type and I change it to static in the UI, I need to send null for the dynamic credential fields to inform the API that we want to change it to static credentials

Reply:
thank you for the explanation!